ios/RNGestureHandler.m

```diff
@@ -370,6 +373,16 @@ - (void)touchesMoved:(NSSet<UITouch *> *)touches withEvent:(UIEvent *)event
         self.state = UIGestureRecognizerStateFailed;
         return;
     }
+    if ((self.state == UIGestureRecognizerStatePossible || self.state == UIGestureRecognizerStateChanged) && _gestureHandler.shouldCancelWhenOutside) {
+        CGPoint pt = [self locationInView:self.view];
+        if (TEST_IS_OUT_OF_BOUNDS(pt, self.view.frame.size)) {
+            self.state = self.state == UIGestureRecognizerStatePossible
```

Comment:
What's the reason you set `Cancelled` if the state was `Changed`? Wouldn't setting `Failed` work regardless? Also calling `reset` shouldn't be necessary (e.g. it is not being called a few lines earlier)

BTW:
One `self.state = ` should be enough :)

Reply:
I'm just being more explicit about the behavior. From what I can tell it isn't possible to go from Changed > Failed in UIGestureRecognizer's state machine. Even if you set Failed here the next state will always be Cancelled if the last state was Changed.